src/sentry/workflow_engine/actions/notification_action/logic.py

```diff
@@ -0,0 +1,137 @@
+import uuid
+from typing import Any
+
+from sentry.constants import ObjectStatus
+from sentry.models.group import GroupEvent
+from sentry.models.rule import Rule, RuleSource
+from sentry.models.rulefirehistory import RuleFireHistory
+from sentry.rules.actions.base import instantiate_action
+from sentry.types.rules import RuleFuture
+from sentry.utils.safe import safe_execute
+from sentry.workflow_engine.actions.notification_action.mappings import (
+    ACTION_TYPE_2_INTEGRATION_ID_KEY,
+    ACTION_TYPE_2_RULE_REGISTRY_ID,
+    ACTION_TYPE_2_TARGET_DISPLAY_KEY,
+    ACTION_TYPE_2_TARGET_IDENTIFIER_KEY,
+)
+from sentry.workflow_engine.models.action import Action
+from sentry.workflow_engine.models.detector import Detector
+
+
+def build_rule_data_blob(action: Action) -> list[dict[str, Any]]:
+    """
+    Builds the Rule.data.actions json blob from the Action model.
+
+    :param action: Action model instance
+    :return: list of dicts with the rule data with length 1
+    """
+
+    # Copy the action data to the rule data
+    rule_data = dict(action.data)
+
+    # Add the rule registry id to the rule data
+    rule_data["id"] = ACTION_TYPE_2_RULE_REGISTRY_ID[action.type]
+
+    # Add the action uuid to the rule data
+    rule_data["uuid"] = action.id
+
+    # Add the integration id to the rule data
+    integration_id_key = ACTION_TYPE_2_INTEGRATION_ID_KEY[action.type]
+    rule_data[integration_id_key] = action.integration_id
+
+    # Add the target identifier to the rule data if it exists
+    target_identifier_key = ACTION_TYPE_2_TARGET_IDENTIFIER_KEY.get(action.type)
+    if target_identifier_key:
+        rule_data[target_identifier_key] = action.target_identifier
+
+    # Add the target display to the rule data if it exists
+    target_display_key = ACTION_TYPE_2_TARGET_DISPLAY_KEY.get(action.type)
+    if target_display_key:
+        rule_data[target_display_key] = action.target_display
+
+    # Because the rule expects "actions" to be a list, we need to return a list with a single dict
+    return [rule_data]
+
+
+def form_issue_alert_models(
```

Comment:
nit: i'd be consistent with these method names so it's clear they're doing similar work. `create_<model>_from_action` is the format i'd use, so it's really clear what is happening.